firmware/CMakeLists.txt

```diff
@@ -220,6 +220,7 @@ function(create_arm_binary
         -Wl,-T ${ARM_LINKER_SCRIPT}
         --specs=nano.specs
     )
+    target_link_libraries(${BOARD_NAME}.elf m) # link math library
```

Comment:
I did not know that [m was the math library](https://stackoverflow.com/questions/34625627/how-to-link-to-the-c-math-library-with-cmake), very interesting namespace choice.